pkg/commands/drift/statefiles/statefiles.go

```diff
@@ -131,6 +178,25 @@ func (c *DriftStatefilesCommand) Process(ctx context.Context) error {
 	logger.DebugContext(ctx, "starting Guardian drift statefiles")
 	logger.DebugContext(ctx, "finding entrypoint directories")
 
+	// Clone all git repositories.
+	repositories, err := c.githubClient.ListRepositories(ctx, c.GitHubFlags.FlagGitHubOwner, &githubAPI.RepositoryListByOrgOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to determine github repositories")
+	}
+	repositoriesWithTerraform := []*github.Repository{}
+	for _, r := range repositories {
+		if len(sets.Subtract(r.Topics, c.flagTerraformRepoTopics)) == 0 {
+			repositoriesWithTerraform = append(repositoriesWithTerraform, r)
+		}
+	}
+
+	for _, r := range repositoriesWithTerraform {
+		if err = c.gitClient.CloneRepository(ctx, c.GitHubFlags.FlagGitHubToken, r.Owner, r.Name); err != nil {
```

Comment:
Seems like this clones the code into the CWD, there COULD be potential collisions with sub-folder names. We should try to target a different directory somehow.

Reply:
My goal is to document this and let the user decide. Then they can `DIR=$(mktmp) && guardian ... $DIR`.

We could instead clone to a tmp dir in the tool. wdyt?

Reply:
I think the tmp dir is better, does that affect your ignore dir logic?

Reply:
This can be a follow up.

Reply:
It does affect it. I'll do it in a follow up.

Reply:
https://github.com/abcxyz/guardian/issues/206